tests/unit/info.tcl

```diff
@@ -465,3 +465,51 @@ start_server {tags {"info" "external:skip"}} {
         }
     }
 }
+
+proc getMemoryStatsProperty {stats property} {
+    set res 0
+    set list_stats [split $stats " "]
+    for {set j 0} {$j < [llength $list_stats]} {incr j} {
+        if {[string match -nocase $property [lindex $list_stats $j]]} {
+            set res [lindex $list_stats [expr $j+1]]
+            break
+        }
+    }
+    return $res
+}
+
+start_server {tags {"info" "external:skip"}} {
+    test {memory: database and pubsub overhead and rehashing dict count} {
+        r flushall
+        set info_mem [r info memory]
+        set mem_stats [r memory stats]
+        assert_equal [getInfoProperty $info_mem mem_overhead_hashtable_rehashing] {0}
+        assert_equal [getMemoryStatsProperty $mem_stats overhead.hashtable.lut] {0}
+        assert_equal [getMemoryStatsProperty $mem_stats overhead.hashtable.rehashing] {0}
+        assert_equal [getMemoryStatsProperty $mem_stats dict.rehashing.count] {0}
+        # Initial dict expand is not rehashing
+        r set a b
+        set info_mem [r info memory]
+        set mem_stats [r memory stats]
+        assert_equal [getInfoProperty $info_mem mem_overhead_hashtable_rehashing] {0}
+        assert_equal [getMemoryStatsProperty $mem_stats overhead.hashtable.lut] {32}
+        assert_equal [getMemoryStatsProperty $mem_stats overhead.hashtable.rehashing] {0}
+        assert_equal [getMemoryStatsProperty $mem_stats dict.rehashing.count] {0}
+        # set 4 more keys to trigger rehashing
+        # get the info within a transaction to make sure the rehashing is not completed
+        r multi 
+        r set b c
+        r set c d
+        r set d e
+        r set e f
+        r info memory
+        r memory stats
+        set res [r exec]
```

Comment:
I think to test the overhead, it is better to do it in a clean and undisturbed server so we can have more precise results. So I wrote the test this way: an independent test, and using transaction to make sure overhead is not changed by active rehashing.

Reply:
i have a feeling that we have too many ways to try to prevent resizing in tests, in most places we use a fork child, but recently we added a DEBUG command (#13043).
now we use MULTI.

considering that for testing these metrics, all you need is a test that can reliably run commands when a dict is in a rehashing state, then since we already have such a test, i thought it makes more sense to add some lines to validate the metrics to an existing test rather than create a new one.

you're argue that in your case it may be easier to predict the exact value of the metrics. it may be right, although i don't want to match their exact value anyway, only make sure they get back to zero when they should (no leak).

anyway, this test isn't too long, we can keep it, and if we have problems with it in the future, we can re-consider the other approach.